taipy/core/config/core_section.py

```diff
@@ -28,9 +28,12 @@ class CoreSection(UniqueSection):
 
     Attributes:
         root_folder (str): Path of the base folder for the taipy application. The default value is "./taipy/"
-        storage_folder (str): Folder name used to store Taipy data. The default value is ".data/". It is used in
+        storage_folder (str): Folder name used to store user data. The default value is ".user_data/". It is used in
```

Comment:
How about "data_folder"?

Reply:
This folder is the entrypoint folder for a user to get and drop data. I prefer `data_folder` instead of `storage_folder`, to be honest, but then it will be a breaking change. Am I correct?
What do you think?

For the default value, I have two remarks:
1. Why not make the default value not start with a `.`. In the end, it is made to be accessed by the user. It is not a hidden folder.
2. Why prefix it by user? Why not just `data`?

Reply:
> This folder is the entrypoint folder for a user to get and drop data. I prefer data_folder instead of storage_folder, to be honest, but then it will be a breaking change. Am I correct?

Yes, I kept the config name the same to avoid a breaking change. My propose is to keep storage folder for the user data, so it doesn't have impact right now. We could change the name in 4.0, in the future.

> Why not make the default value not start with a .. In the end, it is made to be accessed by the user. It is not a hidden folder.

I guess theres no reason to keep it hidden.

>Why prefix it by user? Why not just data?

I think data may be a little to generic and could conflict with other things the user have. We do delete stuff from the data folder in some cases, so it may be more secure if it's not a common name.